server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -178,15 +181,21 @@ Response newResponse(
      * @param request      the node-level request
      * @param shardRouting the shard on which to execute the operation
      * @param task         the task for this node-level request
+     * @param nodeContext  the context created by {{@link #createNodeContext()}}
      * @param listener     the listener to notify with the result of the shard-level operation
      */
     protected abstract void shardOperation(
         Request request,
         ShardRouting shardRouting,
         Task task,
+        @Nullable NodeContext nodeContext,
         ActionListener<ShardOperationResult> listener
     );
 
+    protected NodeContext createNodeContext() {
```

Comment:
nit (comment + `@Nullable`)

```suggestion
    /**
     * @return an (optional) node-level context for this operation, passed to each call to {@link #shardOperation}.
     */
    @Nullable
    protected NodeContext createNodeContext() {
```